UI/Panels/Settings/MobiFlightPanel.cs

```diff
@@ -565,8 +565,31 @@ private void addDeviceTypeToolStripMenuItem_Click(object sender, EventArgs e)
                             throw new MaximumDeviceNumberReachedMobiFlightException(MobiFlightLcdDisplay.TYPE, tempModule.Board.ModuleLimits.MaxLcdI2C);
                         }
 
+                        // Check and see if any I2CPins exist in the board definition file. If not then there's no way to add an LCD device
+                        // so throw an error.
+                        var I2CPins = tempModule.Board.Pins.FindAll(x => x.isI2C);
+
+                        if (!(I2CPins?.Any() ?? false))
+                        {
+                            throw new I2CPinsNotDefinedException(MobiFlightLcdDisplay.TYPE);
+                        }
+
+                        // Fix for issue #900. Check for any I2C pins that might already be in use by
+                        // other modules configured for the device.
+                        var pinsInUse = getVirtualModuleFromTree().GetPins(false, true).Where(x => x.Used);
+
+                        // Check and see if any I2C pins are in use. This only looks for the first I2C pin that's
+                        // in use since that's sufficient to throw an error and tell the user what to do. Trying to
+                        // write an error message that works for one or more in use I2C pins is way more trouble
+                        // than it's worth.
+                        var inUseI2CPin = I2CPins.Find(x => pinsInUse?.Contains(x) ?? false);
```

Comment:
The `?.` and `??` handle the case where there are no pins in use on the board at all. in that situation `pinsInUse` is null.

Reply:
Ugly syntax "??" - Contains(x) can be true or false if pinsInUse is not null... and if null, then it returns false. Later in 586 you test for non null... which a boolean always is.

Weird to read this part of the code.

Reply:
Test should simply be 

if (IsAnyI2CPinAlreadyInUse)

see naming proposal earlier.

Reply:
It's not ugly, it's Linq ;)

`var firstInUseI2CPin = availableI2Cpins.Find(x => pinsInUse?.Contains(x) ?? false);` returns the first used pin or null if no I2C pins are in use, it's not a boolean return value. The part in the `Find()` clause is the test used to determine whether a given I2C pin is in the list of used pins.

The `??` works in conjunction with the `?.` to handle the case where no pins are currently in use on the board. Without that the `.Contains(x)` would throw a null reference exception.

The return result isn't a boolean, it's the first I2C pin that is in use by something else or null.